torch/distributed/rpc/api.py

```diff
@@ -65,7 +66,8 @@ def _init_rpc(
     self_name=None,
     self_rank=-1,
     worker_name_to_id=None,
-    num_send_recv_threads=4,
+    num_send_recv_threads=DEFAULT_NUM_SEND_RECV_THREADS,
+    rpc_timeout=DEFAULT_RPC_TIMEOUT,
```

Comment:
If these defaults are set someplace else (in C++ land), why override them here?

It's risky, because you can easily end up with 2 sets of defaults, and then you never know which is in effect.

Reply:
This is true, right now we'd have to manually make sure they stay in sync. One issue is, I'd want to have default arguments for this in python as well, otherwise, the user will always have to pass it in which is cumbersome. I guess with this particular method `_init_rpc` we don't need to override with defaults since it is only called by `_init_model_parallel`, though, so we can remove the defaults here.

Reply:
I think we can try first put all timeout defaults in Python world. 

C++ constructors always take a known timeout.